media/com_patchtester/js/patchtester.js

```diff
@@ -9,48 +9,34 @@ if (typeof Joomla === 'undefined') {
     throw new Error('PatchTester JavaScript requires the Joomla core JavaScript API')
 }
 
-document.addEventListener("DOMContentLoaded", function (event) {
-    const submitPatch = document.querySelectorAll(".submitPatch");
+Joomla.submitbutton = (task) => {
+  if (task !== 'reset' || confirm(Joomla.JText._('COM_PATCHTESTER_CONFIRM_RESET', 'Resetting will attempt to revert all applied patches and removes all backed up files. This may result in a corrupted environment. Are you sure you want to continue?'))) {
+    Joomla.submitform(task);
+  }
+};
+
+const PatchTester = {
+  /**
+   * Process the patch action
+   *
+   * @param {String} task The task to perform
+   * @param {Number} id   The item ID
+   */
+  submitpatch: (task, id) => {
+    document.getElementById('pull_id')?.value = id;
```

Comment:
@dgrammatiko This line is throwing an `Uncaught SyntaxError: invalid assignment left-hand side`. I think the `?` can go because the `pull_id` field should always be present on the page anyway. What do you say?

Reply:
Hmm the `?.` cannot be on the left part on an assignment line. Either remove it altogether or have a simple conditional to throw if the element is not found. I went with the second

Reply:
fwiw these 2 are the same but supported by different ECMASCRIPT versions:
es6 safe: 

```js
 submitpatch: (task, id) => {
    const el =  document.getElementById('pull_id');

    if (el) {
      el.value = id;
    }
   Joomla.submitform(task);
  }

```

[ES2020](https://caniuse.com/mdn-javascript_operators_optional_chaining):

```js
 submitpatch: (task, id) => {
    const value =  document.getElementById('pull_id')?.value;

    if (value) {
     value = id;
    }
   Joomla.submitform(task);
  }

```

I think ES2020 is safe as many J4,J5 already have similar code, but your choice here

Reply:
@dgrammatiko The ES2020 is fine as far as I am concerned because the Patchtester is not for the general public anyway.

The code does not look right. You cannot assign a value to a const. Further I think that `value = id` won't do anything as `value` is just a string and not an element.

Reply:
`value` in the case of ES2020 is the property setter/getter. If it exists then the assignment is doing the setter thing (=), similar to `el.value = id`. So it should be ok